### PR TITLE
Upgrade to Sanic 21

### PR DESCRIPTION
**Pull request by kay0u**

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 from concurrent.futures._base import CancelledError
 from asyncio import Task
 
-import ujson
```
Comment (kay0u) [inline run.py:23]: ujson doesn't convert datetime to json anymore

File: run.py [hunk 2]
```diff
@@ -464,16 +470,12 @@ async def broadcast(message, channels):
...
         for ws in ws_list:
             try:
-                await ws.send(ujson.dumps(message))
+                await ws.send(json.dumps(message, default=datetime_to_epoch_json_converter))
             except ConnectionClosed:
-                dead_ws.append(ws)
-
-    for to_remove in dead_ws:
-        ws_list.remove(to_remove)
+                ws_list.remove(ws)
```
Comment (Psycojoker) [inline run.py:478]: this will crash because python raise an exception if a list has been modified during iteration iirc (except if this behavior has changed)

File: run.py [hunk 3]
```diff
@@ -560,21 +563,21 @@ async def ws_index(request, websocket):
...
             "action": "init_jobs_stream",
             "data": chunk,
-        }))
+        }, default=datetime_to_epoch_json_converter))
 
     await websocket.wait_closed()
 
 
-@app.websocket('/job-<job_id>-ws')
+@app.websocket('/job-ws/<job_id:int>')
```
Comment (Psycojoker) [inline run.py:580]: Why have you made this change? It doesn't look related to upgrading but maybe am missing something '-'

Comment (kay0u) [reply run.py:580]: I'm not sure why, but routes with a parameter in the middle are never reached. Perhaps you have a better understanding?

File: run.py (hunk 2 above)
Comment (kay0u) [reply run.py]: :+1:

File: run.py (hunk 3 above)
Comment (Psycojoker) [reply run.py:583]: ah, idk, it was working, maybe that's a new thing?

Comment (kay0u) [reply run.py:583]: I didn't find anything about this.

**Review by alexAubin — APPROVED** [review]

**Comment by alexAubin** [comment]
So uuuuh shall we merge this ? :thinking:
